docs/deployment.md

```diff
@@ -27,225 +28,301 @@ Configuration of the PDP is accomplished through two mechanisms:
-
-1. We could let the GitHub Docker publishing action create a new docker image for us each time we commit a change, but that is slow, consumes a lot resources unnecessarily (including limited Dockerhub pulls), and requires a public commit before you may be ready to commit. 
-
-1. Instead we can build and run locally.
-
-#### How
-
-***0 - Advance prep***
-
-1. Update `docker/dev-local/fe_deployment.env` and `docker/dev-local/be_deployment.env` with correct passwords for the `pcic_meta` and `crmp` databases.
-1. Update  `docker/dev-local/pgbounce_users.txt` with correct md5 sums.
-2. Edit your `/etc/hosts` and add `pdp.localhost` to the line starting with `127.0.0.1`. The result will look like 
-```
-127.0.0.1       localhost pdp.localhost
-```
-This allows the reverse proxy automatically set up by the docker-compose to refer to the domain `pdp.localhost`. Note that the frontend container is configured with `APP_ROOT` and `DATA_ROOT` using this domain.
-
-***1 - Build the dev local image***
-
-The image need only be (re)built when:
-
-1. the project is first cloned, or
-2. any of the `*requirements.txt` files change, or
-3. `entrypoint.sh` changes.
-
-The built image contains all dependencies specified in those files (but not the PDP codebase). It forms the basis for installing and running your local codebase.
-
-To build the image:
-
-```
-docker-compose -f docker/dev-local/docker-compose.yaml build
-```
-
-Image build can take several minutes.
+## Deploying locally for development
+
+### Introduction
+
+It is difficult if not impossible to install the PDP on a typical development 
+workstation (particularly since the transition to Ubuntu 20.04). 
+
+To fill that gap, we've defined Docker infrastructure that allows you to 
+build and run a development deployment of the PDP on your workstation 
+that is equivalent to the production deployment. The infrastructure is in 
+`docker/local-run/`.
+
+The infrastructure creates 4 docker containers:
+
+- PDP frontend (`pdp-local-run-fe`)
+- PDP backend (`pdp-local-run-be`)
+- PGbouncer (`pgbouncer-dev`): Manages connections to the databases.
+- Nginx proxy (`pdp-dev-proxy`): Proxies frontend and backend containers to
+  a single domain (`pdp.localhost:<port>`) to avoid CORS issues.
+  
+The frontend and backend containers mount your local codebase and install it,
+so that changes you make dynamically to the code are reflected inside the
+container as you work.
+
+The running app is available at http://pdp.localhost:5000/portal/<name>/map/;
+for example, http://pdp.localhost:5000/portal/bc_prism/map/.
+
+### Instructions
+
+1. **Advance prep**
+
+    Do each of the following things *once per workstation*.
+    You may already have done them previously, for this service or
+    another one.
+    
+    1. Set up user mapping in your local Docker daemon. 
```

Comment:
Might be useful to explicitly note you need to get the other repository here, instead of trying to follow directions in the other repository with this repository, which is what I ended up doing.